### PR TITLE
Converted merge-folders.js to Ramda style

### PR DESCRIPTION
**Pull request by meszaros-lajos-gyorgy**
If we are already using Ramda.js, then we can use it throughout the whole file to have a more functional code base. This will make us concentrate more on what is being done instead of how and reason about where to split the code into reusable parts.

I've translated the code into a pointfree style and where it was not possible, I changed the codes to an arrow syntax. All functions support currying.

### CODE REVIEW

File: src/merge-folders.js [hunk 1]
```diff
@@ -3,53 +3,55 @@
...
-      if (parentPath.endsWith('**') && relative.split(path.sep)[1] !== '..') {
-        retValue = true
-      }
-    } else {
-      retValue = true
-    }
-  }
+      R.compose(
+        R.and(endsWith('**')(parentPath)),
+        R.compose(R.not, R.equals('..'), R.nth(1), R.split(path.sep))
```
Comment (bahmutov) [inline src/merge-folders.js]: can this be separated into a named variable? I am getting confused what it does just by looking at it

**Review by bahmutov — COMMENTED** [review]
I like it, but are the functions unit tested to safely refactor?

**Comment by meszaros-lajos-gyorgy** [comment]
I used ```mocha src/*-spec.js --watch``` to validate every step I make during refactoring. Also I have managed to do a fair bit of cleanup at the point, where you added the comment. It actually improves the readability compared to the original imperative code.

**Comment by meszaros-lajos-gyorgy** [comment]
TODO: need to refactor **appendDoubleStars()**.
TODO2: need to write unit tests for the new functions.

**Comment by bahmutov** [comment]
These are pretty good changes, thanks for takin time to make it more readable @meszaros-lajos-gyorgy 

I am going to add you as a maintainer, because it seems like you care, ok? So you can merge the code yourself.

One thing - this project uses commit syntax to trigger automatic NPM publish on semantic commits (https://glebbahmutov.com/blog/how-i-publish-to-npm/) FYI

**Comment by meszaros-lajos-gyorgy** [comment]
Thank you for the privilege! I will work a bit more on my todos in this ticket and then I'll merge it.

**Comment by meszaros-lajos-gyorgy** [comment]
@bahmutov can you check the code again to see if everything looks good to you?

File: src/merge-folders-spec.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
+const path = require('path')
+const assert = require('assert')
+
+const isFunction = x => typeof x === 'function'
```
Comment (bahmutov) [inline src/merge-folders-spec.js:4]: why taking `check-more-types` out just to write this little function then? Why not use `R.is` then?

**Comment by bahmutov** [comment]
In general I am fine with this, but would prefer smaller changes, this is a large change with many dependencies moved around. Please try to commit smaller things, this could break a lot of people accidentally.

File: src/merge-folders-spec.js (hunk 2 above)
Comment (meszaros-lajos-gyorgy) [reply src/merge-folders-spec.js:4]: I didn't want to load any libraries just to use one function from it, especially if the function is as simple as this. If I would do more type checks, then I would definietly use check-more-types.

**Comment by meszaros-lajos-gyorgy** [comment]
I can create multiple smaller pull requests, if you would like to. We can close this one and I'll try splitting the code.

**Pull request closed without merging**